src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/induction/QualificationsAndTrainingResourceMapper.kt

```diff
@@ -42,24 +52,30 @@ abstract class QualificationsAndTrainingResourceMapper {
 
   fun toEducationAndQualificationResponse(
     qualifications: PreviousQualifications?,
-    training: PreviousTraining?,
-  ): EducationAndQualificationResponse? {
-    return training?.let {
-      EducationAndQualificationResponse(
-        // We always expect to have a value within PreviousTraining (even if it only contains just TrainingType.NONE),
-        // so we use PreviousTraining as the parent/primary object. However, we are still guarding against it being
-        // null in case anything changes on the client side.
-        id = training.reference,
-        educationLevel = toHighestEducationLevelApi(qualifications?.educationLevel),
-        qualifications = toAchievedQualifications(qualifications?.qualifications),
-        additionalTraining = toTrainingTypesApi(training.trainingTypes),
-        additionalTrainingOther = if (training.trainingTypes.contains(OTHER)) training.trainingTypeOther else null,
-        modifiedBy = training.lastUpdatedBy!!,
-        modifiedDateTime = toOffsetDateTime(training.lastUpdatedAt)!!,
-      )
+    training: PreviousTraining,
+  ): EducationAndQualificationResponse {
+    // We always expect to have a value within training (even if it only contains TrainingType.NONE), so we use that
+    // as the "primary" object from which to populate the JPA generated fields from. However, if qualifications have
+    // been updated more recently, then we populate modifiedDateTime and modifiedBy from there instead.
```

Comment:
I think this is right because if I send in a request to update the list of qualifications, but not the training, then I would expect the modifiedDate of the `qualificationsAndTraining` object in the response to have an updated timestamp.